index.js

```diff
@@ -1,11 +1,68 @@
-var _ = require('lodash')
-var async = require('async')
-var exec = require('child_process').exec
-var gutil = require('gulp-util')
-var path = require('path')
-var through = require('through2')
+var _ = require('lodash');
+var async = require('async');
+var exec = require('child_process').exec;
+var gutil = require('gulp-util');
+var path = require('path');
+var throughConcurrent = require('through2-concurrent');
+
+var PLUGIN_NAME = 'gulp-async-shell'
+
+// Returns a function that async.parallel will use, each function is its own shell command
+function genShellExecFunction(command,file,options,callback){
+  var context = _.extend({file: file}, options.templateData)
+  command = gutil.template(command, context)
+
+  var child = exec(command, {
+    env: options.env,
+    cwd: options.cwd,
+    maxBuffer: options.maxBuffer,
+    timeout: options.timeout
+  }, function (error, stdout, stderr) {
+    process.stdin.unpipe(child.stdin)
+    process.stdin.resume()
+    process.stdin.pause()
+
+    if (error && !options.ignoreErrors) {
+      error.stdout = stdout
+      error.stderr = stderr
+
+      var errorContext = _.extend({
+        command: command,
+        file: file,
+        error: error
+      }, options.templateData)
+
+      error.message = gutil.template(options.errorMessage, errorContext)
+    }
+    if(!options.quiet && options.logMessage){
+      var date = new Date()
+      var hours = date.getHours()
+      var hourStr = hours <= 9 ? "0"+hours : ""+hours
+      var time = hourStr+":"+date.getMinutes()+":"+date.getSeconds()
```

Comment:
You probably wanted to zero pad the minutes and seconds as well.  Here's a cleaner idiom
```
var time = ('0' + hours).slice(-2) + ':' +  ('0' + date.getMinutes()).slice(-2) + ':' +  ('0' + date.getSeconds()).slice(-2);
```